### PR TITLE
Fix submodelrepository mongodb backend

### PR DESCRIPTION
**Pull request by msiebert-dev**
This pull request fixes the submodelrepository not using a mongoDB backend when basyx.backend=MongoDB in application.properties.

### CODE REVIEW

File: basyx.submodelrepository/basyx.submodelrepository-backend-mongodb/src/main/java/org/eclipse/digitaltwin/basyx/submodelrepository/MongoDBSubmodelRepositoryFactory.java [hunk 1]
```diff
@@ -30,29 +30,34 @@
...
+	@Autowired(required = false)
 	public MongoDBSubmodelRepositoryFactory(MongoTemplate mongoTemplate, @Value("${basyx.submodelrepository.mongodb.collectionName:submodel-repo}") String collectionName, SubmodelServiceFactory submodelServiceFactory) {
 		this.mongoTemplate = mongoTemplate;
 		this.collectionName = collectionName;
 		this.submodelServiceFactory = submodelServiceFactory;
 	}
 
-	@Autowired
+
+	@Autowired(required = false)
```
Comment (FrankSchnicke) [inline basyx.submodelrepository/basyx.submodelrepository-backend-mongodb/src/main/java/org/eclipse/digitaltwin/basyx/submodelrepository/MongoDBSubmodelRepositoryFactory.java:60]: Why is "required = false" needed here?

**Review by FrankSchnicke — COMMENTED** [review]
Thank you very much for this contribution! After my remark is clarified, I will gladly merge.

File: basyx.submodelrepository/basyx.submodelrepository-backend-mongodb/src/main/java/org/eclipse/digitaltwin/basyx/submodelrepository/MongoDBSubmodelRepositoryFactory.java (hunk 1 above)
Comment (msiebert-dev) [reply basyx.submodelrepository/basyx.submodelrepository-backend-mongodb/src/main/java/org/eclipse/digitaltwin/basyx/submodelrepository/MongoDBSubmodelRepositoryFactory.java:60]: If "basyx.backend" is set to "MongoDB" in application.properties, the application runs into a BeanCreationException without the required attribute set to false. The attribute's default value is true.
From the [current Spring documentation](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/annotation/Autowired.html): "Only one constructor of any given bean class may declare this annotation with the [required()](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/annotation/Autowired.html#required()) attribute set to true, indicating the constructor to autowire when used as a Spring bean. Furthermore, if the required attribute is set to true, only a single constructor may be annotated with @Autowired. If multiple non-required constructors declare the annotation, they will be considered as candidates for autowiring. The constructor with the greatest number of dependencies that can be satisfied by matching beans in the Spring container will be chosen. [...]"
As both constructors were annotated with @Autowired I chose to set the required attribute to false.

Comment (FrankSchnicke) [reply basyx.submodelrepository/basyx.submodelrepository-backend-mongodb/src/main/java/org/eclipse/digitaltwin/basyx/submodelrepository/MongoDBSubmodelRepositoryFactory.java:60]: Thank you very much for clarifying.

**Pull request merged**